app/plugins/Files/js/buttons.js

```diff
@@ -127,6 +129,43 @@ eID('cancel-ascii').onclick = function() {
 	hide('show-ascii');
 };
 
+// Select directory sliding frame
+eID('new-dir').onclick = function() {
+	show('add-dir');
+	hide('file-library');
+};
+eID('back-dir').onclick = exitFileAdder;
+
+// Upload directory option chosen
+eID('upload-dir-choice').onclick = function() {
+	var path = require("path");
+	hideSteps([21,31]);
+
+	var loadPath = IPC.sendSync('dialog', 'open', {
+		title: 'Select Directory',
+		properties: ['openDirectory'],
+	});
+	if (loadPath) {
+		eID('nickname-dir').querySelector('.dir-path').innerHTML = loadPath;
+		loadPath = loadPath[0].split(path.sep);
+		eID('nickname-dir-input').value = loadPath[loadPath.length - 1] + '_';
+		show('nickname-dir');
+		show('upload-dir');
+		eID('nickname-dir-input').focus();
+	}
+};
+
+eID('upload-dir').onclick = function() {
+	var loadPath = eID('nickname-dir').querySelector('.dir-path').innerHTML;
+	var nickname = eID('nickname-dir-input').value;
+	// Illegal filename characters in nickname seems to throw errors
+	// So, substitute \ and / with underscore (_)
+	nickname.replace(/[/\\\\]/g, "_");
```

Comment:
The gets a prefix, which is the folder containing the files: `C:\\Users\Ming\Documents\test.txt` becomes `Documents_test.txt` by default. The user can edit the prefix (ie `Documents_`) before uploading. The reason for the substitution was errors when `/` was included in nicknames. For future reference, it would be nice to allow `/` and `\` in nicknames as one could then display a virtual file structure in the Sia-UI.